sympy/printing/maple.py

```diff
@@ -0,0 +1,220 @@
+    'Abs': 'abs',
+    'log': 'ln',
+    'asin': 'arcsin',
+    'acos': 'arccos',
+    'atan': 'arctan',
+    'asec': 'arcsec',
+    'acsc': 'arccsc',
+    'acot': 'arccot',
+    'asinh': 'arcsinh',
+    'acosh': 'arccosh',
+    'atanh': 'arctanh',
+    'asech': 'arcsech',
+    'acsch': 'arccsch',
+    'acoth': 'arccoth',
+    'ceiling': 'ceil',
+}
+for _func in _known_func_same_name:
+    known_functions[_func] = _func
+
+number_symbols = {
+    # Sympy -> Maple
+    S.Pi: 'Pi',
+    S.Exp1: 'exp(1)',
+    S.Catalan: 'Catalan',
+    S.EulerGamma: 'gamma'
+}
+
+spec_relational_ops = {
+    # Sympy -> Maple
+    '==': '=',
+    '!=': '<>'
+}
+
+
+class MapleCodePrinter(CodePrinter):
+    """
+    Printer which converts single sympy expressions into single.
+    """
+    printmethod = "_maple"
+    language = "maple"
+
+    def __init__(self, settings=None):
+        if settings is None:
+            settings = dict()
+        super(MapleCodePrinter, self).__init__(settings)
+        self.known_functions = dict(known_functions)
+        userfuncs = settings.get('user_functions', {})
+        self.known_functions.update(userfuncs)
+
+    def _get_statement(self, codestring):
+        return "%s;" % codestring
+
+    def _get_comment(self, text):
+        return "# {0}".format(text)
+
+    def _declare_number_const(self, name, value):
+        return "{0} := {1};".format(name,
+                                    value.evalf(self._settings['precision']))
+
+    def _format_code(self, lines):
+        return lines
+
+    def _print_Pow(self, expr):
+        PREC = precedence(expr)
+        if expr.exp == -1:
+            return '1/%s' % (self.parenthesize(expr.base, PREC))
+        elif expr.exp == 0.5 or expr.exp == S(1) / 2:
+            return 'sqrt(%s)' % self._print(expr.base)
+        elif expr.exp == -0.5 or expr.exp == -S(1) / 2:
+            return '1/sqrt(%s)' % self._print(expr.base)
+        else:
+            return '%s^%s' % (self._print(expr.base), self._print(expr.exp))
+
+    def _print_Rational(self, expr):
+        p, q = int(expr.p), int(expr.q)
+        return '%d/%d' % (p, q)
+
+    def _print_Relational(self, expr):
+        lhs_code = self._print(expr.lhs)
+        rhs_code = self._print(expr.rhs)
+        op = expr.rel_op
+        if op in spec_relational_ops:
+            op = spec_relational_ops[op]
+        return "{0} {1} {2}".format(lhs_code, op, rhs_code)
+
+    def _print_NumberSymbol(self, expr):
+        return number_symbols[expr]
+
+    def _print_NegativeInfinity(self, expr):
+        return '-infinity'
+
+    def _print_Infinity(self, expr):
+        return 'infinity'
+
+    def _print_Idx(self, expr):
+        return self._print(expr.label)
+
+    def _print_BooleanTrue(self, expr):
+        return "true"
+
+    def _print_BooleanFalse(self, expr):
+        return "false"
+
+    def _print_bool(self, expr):
+        return str(expr).lower()
+
+    def _print_NaN(self, expr):
+        return 'NULL'
```

Comment:
Should it be 'undefined' rather than NULL?
https://www.maplesoft.com/support/help/Maple/view.aspx?path=undefined

Reply:
Yes if NaN is a result of 1/0, then it should be undefined rather than NULL.